utils/s2n_socket.c

```diff
@@ -192,7 +192,6 @@ int s2n_socket_read(void *io_context, uint8_t *buf, uint32_t len)
 
     /* On success, the number of bytes read is returned. On failure, -1 is
      * returned and errno is set appropriately. */
-    errno = 0;
```

Comment:
We don't technically check the syscall return value here, we rely that the caller does the right thing.

Reply:
We also don't actually do anything with errno here though :) It's up to whoever needs to use errno to make sure they use it right. The errno system is kind of built on the caller doing the right thing :\ It's dangerous to assume any function sets errno to zero on success, because no system calls make that guarantee. 

I can leave the errno=0s in here, but they encourage a non-standard use of errno that assumes a function resets it. If the caller cares about errno, they have to either reset it before the call or check the return value. I'd argue for always resetting when calling an s2n_* function, bc we can't trust s2n internal functions to maintain behavior the way we can trust system calls.

Reply:
Makes sense. Removing them now is better then removing them later when somebody might be relying on this function doing the reset.